src/shared.ts

```diff
@@ -58,3 +58,74 @@ export function isFunction(obj: any): obj is Function {
 }
 
 export const emptyArray: [] = [];
+
+export const ESGlobalKeys = new Set([
+
+    // *** 18.1 Value Properties of the Global Object
+    'Infinity',
+    'NaN',
+    'undefined',
+
+    // *** 18.2 Function Properties of the Global Object
+    'eval', // dangerous
+    'isFinite',
+    'isNaN',
+    'parseFloat',
+    'parseInt',
+    'decodeURI',
+    'decodeURIComponent',
+    'encodeURI',
+    'encodeURIComponent',
+
+    // *** 18.3 Constructor Properties of the Global Object
+    'Array',
+    'ArrayBuffer',
+    'Boolean',
+    'DataView',
+    'Date', // Unstable
+    'Error', // Unstable
+    'EvalError',
+    'Float32Array',
+    'Float64Array',
+    'Function', // dangerous
+    'Int8Array',
+    'Int16Array',
+    'Int32Array',
+    'Map',
+    'Number',
+    'Object',
+    'Promise', // Unstable
+    'Proxy', // Unstable
+    'RangeError',
+    'ReferenceError',
+    'RegExp', // Unstable
+    'Set',
+    // 'SharedArrayBuffer', // removed on Jan 5, 2018
+    'String',
+    'Symbol',
+    'SyntaxError',
+    'TypeError',
+    'Uint8Array',
+    'Uint8ClampedArray',
+    'Uint16Array',
+    'Uint32Array',
+    'URIError',
+    'WeakMap',
+    'WeakSet',
+
+    // *** 18.4 Other Properties of the Global Object
+
+    // 'Atomics', // removed on Jan 5, 2018
+    'JSON',
+    'Math',
+    'Reflect',
+
+    // *** Annex B
+
+    'escape',
+    'unescape',
+
+    // *** ECMA-402
+
+    'Intl', // Unstable
+]);
```

Comment:
Possibly more to consider [here](https://github.com/nodejs/node/blob/master/lib/internal/freeze_intrinsics.js).